debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlTinyIntIT.java

```diff
@@ -0,0 +1,94 @@
+import org.apache.kafka.connect.source.SourceRecord;
+import org.fest.assertions.Assertions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.debezium.config.Configuration;
+import io.debezium.doc.FixFor;
+import io.debezium.embedded.AbstractConnectorTest;
+import io.debezium.util.Testing;
+
+/**
+ * Verify correct range of TINYINT.
+ *
+ * @author Jiri Pechanec
+ */
+public class MySqlTinyIntIT extends AbstractConnectorTest {
+
+    private static final Path DB_HISTORY_PATH = Testing.Files.createTestingPath("file-db-history-year.txt")
+            .toAbsolutePath();
+    private final UniqueDatabase DATABASE = new UniqueDatabase("tinyintit", "tinyint_test")
+            .withDbHistoryPath(DB_HISTORY_PATH);
+
+    private Configuration config;
+
+    @Before
+    public void beforeEach() {
+        stopConnector();
+        DATABASE.createAndInitialize();
+        initializeConnectorTestFramework();
+        Testing.Files.delete(DB_HISTORY_PATH);
+    }
+
+    @After
+    public void afterEach() {
+        try {
+            stopConnector();
+        }
+        finally {
+            Testing.Files.delete(DB_HISTORY_PATH);
+        }
+    }
+
+    @Test
+    @FixFor("DBZ-1773")
+    public void shouldHandleTinyIntAsNumber() throws SQLException, InterruptedException {
+        // Use the DB configuration to define the connector's configuration ...
+        config = DATABASE.defaultConfig()
+                .with(MySqlConnectorConfig.SNAPSHOT_MODE, MySqlConnectorConfig.SnapshotMode.INITIAL)
+                .build();
+
+        // Start the connector ...
+        start(MySqlConnector.class, config);
+
+        // ---------------------------------------------------------------------------------------------------------------
+        // Consume all of the events due to startup and initialization of the database
+        // ---------------------------------------------------------------------------------------------------------------
+        final int numDatabase = 2;
+        final int numTables = 2;
+        final int numOthers = 2;
+        consumeRecords(numDatabase + numTables + numOthers);
+
+        assertChangeRecord();
+
+        try (final Connection conn = MySQLConnection.forTestDatabase(DATABASE.getDatabaseName()).connection()) {
+            conn.createStatement().execute("INSERT INTO DBZ1773 VALUES (DEFAULT, 100, 5, 50)");
+        }
+        assertChangeRecord();
+
+        stopConnector();
+    }
+
+    private void assertChangeRecord() throws InterruptedException {
+        final SourceRecord record = consumeRecord();
+        Assertions.assertThat(record).isNotNull();
+        final Struct change = ((Struct) record.value()).getStruct("after");
+
+        Assertions.assertThat(change.getInt16("ti")).isEqualTo((short) 100);
+        Assertions.assertThat(change.getInt16("ti1")).isEqualTo((short) 5);
+        Assertions.assertThat(change.getInt16("ti2")).isEqualTo((byte) 50);
```

Comment:
How comes the expected type differs between short and byte? As per the docs, it should always be `int8`? The `N` in `TINYINT(N)` is only the display width, it doesn't impact the value range AFAIU.

Reply:
@jpechane, any thoughts on this?

Reply:
@gunnarmorling Sorry, was a mistake on my side, fixed. In fact it does not matter if value passed is short or byte as it is alway casted to short according to method parameter. But good to align for consistency

Reply:
Ok, thanks!